src/util/simplify_expr2.cpp

```diff
@@ -415,6 +415,258 @@ static type2tc common_arith_op2_type(expr2tc &e, expr2tc &f)
 
 expr2tc add2t::do_simplify() const
 {
+  // X + X --> X << 1
+  if(is_symbol2t(side_1) && is_symbol2t(side_2))
+  {
+    if(to_symbol2t(side_1) == to_symbol2t(side_2))
+      return shl2tc(type, side_1, from_integer(1, type));
+  }
+
+  // (A + 1) + ~B --> A - B
+  auto simplify_1 = [](const expr2tc e, const expr2tc f) -> expr2tc {
+    auto sidecheck = to_add2t(e).side_2;
+    auto sidecheck2 = to_add2t(e).side_1;
```

Comment:
Variable naming is somewhat confusing: `side_1` gets called `sidecheck2`. Why is a new name needed if it is only used once? Also another copy.